api/src/test/java/org/openmrs/module/patientqueueing/api/PatientQueueingServiceTest.java

```diff
@@ -158,4 +165,92 @@ public void savePatientQueue_shouldCompleteExistingPatientQueueWithSamePatientAn
 		Assert.assertEquals(PatientQueue.Status.COMPLETED, patientQueue1.getStatus());
 		
 	}
+	
+	@Test
+	public void generateVisitNumber_shouldReturnVisitNumberBasedOnPatientAndLocation() {
+		PatientQueueingService patientQueueingService = Context.getService(PatientQueueingService.class);
+		
+		Patient patient = Context.getPatientService().getPatient(10000);
+		
+		Location location = Context.getLocationService().getLocation(1);
+		
+		String visitNumber = patientQueueingService.generateVisitNumber(location, patient);
+		
+		SimpleDateFormat formatter = new SimpleDateFormat(Context.getAdministrationService().getGlobalProperty(
+		    "patientqueueing.defaultDateFormat"));
+		
+		Assert.assertEquals(formatter.format(new Date()) + "-Unk" + "-001", visitNumber);
+		
+	}
+	
+	@Test
+	public void assignVisitNumber_shouldAssignPatientQueueNewVisitNumberWhenNoPatientQueueExistsOnSameDate() {
+		PatientQueueingService patientQueueingService = Context.getService(PatientQueueingService.class);
+		
+		Patient patient = Context.getPatientService().getPatient(10000);
+		
+		Location location = Context.getLocationService().getLocation(1);
+		
+		PatientQueue patientQueue = new PatientQueue();
+		patientQueue.setPatient(patient);
+		patientQueue.setStatus(PatientQueue.Status.PENDING);
+		patientQueue.setEncounter(Context.getEncounterService().getEncounter(10000));
+		patientQueue.setLocationFrom(location);
+		patientQueue.setLocationTo(location);
+		patientQueue.setPriority(0);
+		patientQueue.setPriorityComment("Emergency");
+		patientQueueingService.assignVisitNumber(patientQueue);
+		
+		String newQueueNumber = patientQueueingService.generateVisitNumber(location, patient);
+		
+		String visitNumber = patientQueueingService.generateVisitNumber(location, patient);
+		
+		Assert.assertEquals(newQueueNumber, visitNumber);
```

Comment:
But the test name says it should assign new visit number. Should you be asserting equals?